core/lib/spree/core/permalinks.rb

```diff
@@ -44,21 +44,18 @@ def permalink_order
       end
 
       def generate_permalink
-        "#{self.class.permalink_prefix}#{Array.new(self.class.permalink_length){rand(9)}.join}"
+        permalink_value = "#{self.class.permalink_prefix}#{Array.new(self.class.permalink_length){rand(9)}.join}"
       end
 
-      def save_permalink(permalink_value=self.to_param)
+      def save_unique_permalink(permalink_value=self.to_param)
         self.with_lock do
           permalink_value ||= generate_permalink
-
           field = self.class.permalink_field
             # Do other links exist with this permalink?
             other = self.class.where("#{self.class.table_name}.#{field} LIKE ?", "#{permalink_value}%")
             if other.any?
-              # Find the existing permalink with the highest number, and increment that number.
-              # (If none of the existing permalinks have a number, this will evaluate to 1.)
-              number = other.map { |o| o.send(field)[/-(\d+)$/, 1].to_i }.max + 1
-              permalink_value += "-#{number.to_s}"
+              permalink_value ||= generate_permalink
```

Comment:
Since this is now recursive, I don't think you will need this line (it should be called on each subsequent call of `save_unique_permalink`) Let me know if I'm reading this wrong / missing something though